flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetSingleRowJoin.scala

```diff
@@ -144,21 +150,46 @@ class DataSetSingleRowJoin(
 
     val condition = codeGenerator.generateExpression(joinCondition)
 
-    val joinMethodBody = s"""
-                  |${condition.code}
-                  |if (${condition.resultTerm}) {
-                  |  ${conversion.code}
-                  |  ${codeGenerator.collectorTerm}.collect(${conversion.resultTerm});
-                  |}
-                  |""".stripMargin
+    val joinMethodBody =
+      if (joinType == JoinRelType.INNER) {
+        s"""
+         |${condition.code}
+         |if (${condition.resultTerm}) {
+         |  ${conversion.code}
+         |  ${codeGenerator.collectorTerm}.collect(${conversion.resultTerm});
+         |}
+         |""".stripMargin
+    } else {
+        val singleNode =
+          if (leftIsSingle) {
+            leftNode
+          }
+          else {
+            rightNode
+          }
+        val notSuitedToCondition = singleNode
+          .getRowType
+          .getFieldList
+          .map(field => getRowType.getFieldNames.indexOf(field.getName))
+          .map(i => s"${conversion.resultTerm}.setField($i,null);")
+
+        s"""
+           |${condition.code}
+           |${conversion.code}
+           |if(!${condition.resultTerm}){
+           |${notSuitedToCondition.mkString("\n")}
```

Comment:
I think this part is excessive, because in result code we get:
```java
//{condition.code}
boolean isNull$55 = isNull$45 || isNull$50;
          boolean result$54;
          if (isNull$55) {
            result$54 = false;
          }
          else {
            result$54 = result$44 > result$49;
          }
      //{conversion.code}    
          if (isNull$45) {
            out.setField(0, null);
          }
          else {
            out.setField(0, result$44);
          }  
          if (isNull$50) {
            out.setField(1, null);
          }
          else {
            out.setField(1, result$49);
          }
        //if(!${condition.resultTerm}){ {notSuitedToCondition.mkString("\n")}  
          if(!result$54){
          out.setField(1,null);
          }
```
As result we twice write to 1 field - rewrite value on null. I don't know how do it better.